components/__init__.py

```diff
@@ -247,6 +244,16 @@ def define_components(reqs):
                 commands=isal_build,
                 required_progs=['nasm', 'yasm'],
                 libs=["isal"])
+    reqs.define('isal_crypto',
+                retriever=GitRepoRetriever(
+                        "https://github.com/intel/isa-l_crypto"),
+                commands=['./autogen.sh ',
+                          './configure --prefix=$ISAL_CRYPTO_PREFIX '
+                          '--libdir=$ISAL_CRYPTO_PREFIX/lib',
+                          'make $JOBS_OPT', 'make install'],
+                required_progs=['nasm', 'yasm'],
```

Comment:
While nasm and yasm have overlaps in their abilities, depending on the product they can build it differently.
We should determine specifically which one is required.

Reply:
The actual requirements is that isa-l looks for nasm >= 2.13 and if nasm fails that test it uses yasm.

Reply:
This would be similar to what we do with stdatomic.h for openpa....just adding a configure check though a little more complex since it needs a version as well.  Probably similar to what we do with go version

Reply:
Ya, I just copied most of isal's definition for isal_crypto. I'll update both to specify the actual requirements better.

Reply:
I haven't figured out how to conditionally specify which asm to use based on if yasm or nasm is installed and which versions, but then I started to wonder if the DAOS build should maintain that information or if it should just rely on the isal build to fail if dependencies aren't met?

With required_progs removed and nasm/yasm removed from system isal has a nice error message:

> checking for yasm... no
no yasm
checking for nasm... no
no nasm
configure: error: No modern yasm or nasm found as required. Yasm should be 1.2.0 or later, and nasm should be v2.11.01 or later (v2.13 for AVX512).
BuildFailure: isal failed to build:
  File "/home/rjensen1/workspace/daos/scons_local/SConstruct", line 53:
    scons()
  File "/home/rjensen1/workspace/daos/scons_local/SConstruct", line 41:
    reqs.require(env, *env.get("REQUIRES"))
  File "/home/rjensen1/workspace/daos/scons_local/prereq_tools/base.py", line 1141:
    raise error

Better than what DAOS currently provides:
> Checking whether nasm program exists...no
MissingSystemLibs: isal has unmet dependancies required for build:
  File "/home/rjensen1/workspace/daos/scons_local/SConstruct", line 53:
    scons()
  File "/home/rjensen1/workspace/daos/scons_local/SConstruct", line 41:
    reqs.require(env, *env.get("REQUIRES"))
  File "/home/rjensen1/workspace/daos/scons_local/prereq_tools/base.py", line 1141:
    raise error

Thoughts?

Reply:
yeah you'd probably have to write a custom configure function.  For now, relying on the underlying configure to tell you is fine IMO

Reply:
we could probably fix that error message to at least provide the name of the dependency that is missing :-)